NonEmptyEnumerable/NonEmptyEnumerable.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace NonEmptyEnumerable
 {
-  public class NonEmptyEnumerable<T> : IReadOnlyCollection<T>, IEnumerable<T> where T : notnull
+  public class NonEmptyEnumerable<T> : IReadOnlyCollection<T>, IEquatable<NonEmptyEnumerable<T>> where T : notnull
```

Comment:
Note for reference:

`IReadOnlyCollection<T>` inherits from `IEnumerable<T>` so explicitly inheriting from both is unnecessary.